gulpfile.js

```diff
@@ -278,23 +229,28 @@ gulp.task('publish', (cb) => {
   let publishedPackages = 0;
   let publishedPackagesSkipped = 0;
 
-  const npmAuth = process.env["npm-auth"];
+  const currentDirectoryPath = __dirname;
+
+  const npmrcFileName = ".npmrc";
+  const npmrcRootFilePath = path.join(currentDirectoryPath, npmrcFileName);
+  const npmrcRootFileExists = fs.existsSync(npmrcRootFilePath);
+  if (npmrcRootFileExists) {
+    console.log(`Found ".npmrc" auth file at "${npmrcRootFilePath}". Using it to authenticate with NPM for publish.`);
+  }
 
   for (let i = 0; i < nodejsReadmeFilePaths.length; ++i) {
     const nodejsReadmeFilePath = nodejsReadmeFilePaths[i];
-    // console.log(`INFO: Processing ${nodejsReadmeFilePath}`);
-
     const nodejsReadmeFileContents = fs.readFileSync(nodejsReadmeFilePath, 'utf8');
     const relativeOutputFolderPath = nodejsReadmeFileContents.match(/output\-folder: \$\(node\-sdks\-folder\)\/(lib\/services\/\S+)/)[1];
     const packageFolderPath = path.resolve(azureSDKForNodeRepoRoot, relativeOutputFolderPath);
     if (!fs.existsSync(packageFolderPath)) {
-      console.log(`ERROR: Package folder ${packageFolderPath} has not been generated.`);
+      console.log(`WARNING: Package folder ${packageFolderPath} has not been generated.`);
       errorPackages++;
     }
     else {
       const packageJsonFilePath = `${packageFolderPath}/package.json`;
       if (!fs.existsSync(packageJsonFilePath)) {
-        console.log(`ERROR: Package folder ${packageFolderPath} is missing its package.json file.`);
+        console.log(`WARNING: Package folder ${packageFolderPath} is missing its package.json file.`);
```

Comment:
not a hard requirement but you can also use console.warn|error|info(). More info over [here](https://nodejs.org/dist/latest-v8.x/docs/api/console.html).

Still keeping the WARNING:  prefix makes sense.